spark/src/test/java/org/apache/iceberg/spark/data/RandomData.java

```diff
@@ -83,9 +84,22 @@ public InternalRow next() {
   }
 
   public static Iterable<Record> generate(Schema schema, int numRecords, long seed) {
+    return newIterable(() -> new RandomDataGenerator(schema, seed), schema, numRecords);
+  }
+
+  public static Iterable<Record> generateFallbackData(Schema schema, int numRecords, long seed, long numDictRecords) {
+    return newIterable(() -> new FallbackDataGenerator(schema, seed, numDictRecords), schema, numRecords);
+  }
+
+  public static Iterable<GenericData.Record> generateDictionaryEncodableData(Schema schema, int numRecords, long seed) {
+    return newIterable(() -> new DictionaryEncodedDataGenerator(schema, seed), schema, numRecords);
+  }
+
+  private static Iterable<Record> newIterable(Supplier<RandomDataGenerator> newGenerator,
+                                              Schema schema, int numRecords) {
```

Comment:
The purpose of these changes is to enable running the tests with more records than we can hold in memory. I'm currently running one with 100,000,000 records to test correctness.